### ISSUE TITLE
30$: report 'unclosed code block' based on indentation

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#282]
In the following code, a human can tell that the `if` statement's `{` is unmatched:
```js
function f() {
  if (true) {
    body();
}
```
quick-lint-js is a stupid computer and says that `f`'s `{` is unmatched.

quick-lint-js should track indentation somehow (at least for the purpose of reporting this error) so it can point to the correct unmatched bracket.
